### ISSUE TITLE
Missing Django requirement?

### ISSUE THREAD

**Issue by gsilvapt** [surface-security/django-environ-ppb#5]
Hi :wave:

While trying to upgrade and convert this repository to use the new conventions, I got [some errors by missing `Django`](https://github.com/surface-security/django-environ-ppb/actions/runs/4196265728/jobs/7276992730):

```
PYTHONPATH="testapp" \
	python -b -W always -m coverage run testapp/manage.py test ${TEST_ARGS:-tests}
Traceback (most recent call last):
  File "testapp/manage.py", line 11, in main
    from django.core.management import execute_from_command_line
ModuleNotFoundError: No module named 'django'
```

Looking at this line above and comparing with all other `django-*` apps we have in the org, I believe this app is missing a dependency there. Initially, I thought `django-environ` was requiring Django but checking in [their repo](https://github.com/joke2k/django-environ) it's clear they have no installation requirements either.

https://github.com/surface-security/django-environ-ppb/blob/0f972b7c669e646d4256a6cfebf996e799fb05d2/setup.cfg#L29-L31

After adding `Django >= 3.0, <4` as we have done to others, test suite works successfully.

```
(django-environ-ppb) MACVQTH71XGGF :: github.com/surface-security/django-environ-ppb ‹misc-tweak_workflows*› » tox -e py -v
py: find interpreter for spec PythonSpec(path=/Users/silvag3/.pyenv/versions/3.11.1/envs/django-environ-ppb/bin/python3.11)
py: proposed PythonInfo(spec=CPython3.11.1.final.0-64, system=/Users/silvag3/.pyenv/versions/3.11.1/bin/python3.11, exe=/Users/silvag3/.pyenv/versions/3.11.1/envs/django-environ-ppb/bin/python3.11, platform=darwin, version='3.11.1 (main, Jan 11 2023, 15:59:02) [Clang 14.0.0 (clang-1400.0.29.202)]', encoding_fs_io=utf-8-utf-8)
.pkg: find interpreter for spec PythonSpec(path=/Users/silvag3/.pyenv/versions/3.11.1/envs/django-environ-ppb/bin/python3.11)
.pkg: proposed PythonInfo(spec=CPython3.11.1.final.0-64, system=/Users/silvag3/.pyenv/versions/3.11.1/bin/python3.11, exe=/Users/silvag3/.pyenv/versions/3.11.1/envs/django-environ-ppb/bin/python3.11, platform=darwin, version='3.11.1 (main, Jan 11 2023, 15:59:02) [Clang 14.0.0 (clang-1400.0.29.202)]', encoding_fs_io=utf-8-utf-8)
.pkg: _optional_hooks> python /Users/silvag3/.pyenv/versions/3.11.1/envs/django-environ-ppb/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: exit None (0.13 seconds) /Users/silvag3/github.com/surface-security/django-environ-ppb> python /Users/silvag3/.pyenv/versions/3.11.1/envs/django-environ-ppb/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__ pid=62671
.pkg: get_requires_for_build_sdist> python /Users/silvag3/.pyenv/versions/3.11.1/envs/django-environ-ppb/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: exit None (0.03 seconds) /Users/silvag3/github.com/surface-security/django-environ-ppb> python /Users/silvag3/.pyenv/versions/3.11.1/envs/django-environ-ppb/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__ pid=62671
.pkg: prepare_metadata_for_build_wheel> python /Users/silvag3/.pyenv/versions/3.11.1/envs/django-environ-ppb/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: exit None (0.03 seconds) /Users/silvag3/github.com/surface-security/django-environ-ppb> python /Users/silvag3/.pyenv/versions/3.11.1/envs/django-environ-ppb/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__ pid=62671
.pkg: build_sdist> python /Users/silvag3/.pyenv/versions/3.11.1/envs/django-environ-ppb/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg: exit None (0.02 seconds) /Users/silvag3/github.com/surface-security/django-environ-ppb> python /Users/silvag3/.pyenv/versions/3.11.1/envs/django-environ-ppb/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__ pid=62671
py: install_package_deps> python -I -m pip install --pre 'Django<4,>=3.0' --pre
py: exit 0 (4.17 seconds) /Users/silvag3/github.com/surface-security/django-environ-ppb> python -I -m pip install --pre 'Django<4,>=3.0' --pre pid=62728
py: install_package> python -I -m pip install --pre --force-reinstall --no-deps /Users/silvag3/github.com/surface-security/django-environ-ppb/.tox/.tmp/package/2/django-environ-ppb-1.0.0.tar.gz --pre
py: exit 0 (0.62 seconds) /Users/silvag3/github.com/surface-security/django-environ-ppb> python -I -m pip install --pre --force-reinstall --no-deps /Users/silvag3/github.com/surface-security/django-environ-ppb/.tox/.tmp/package/2/django-environ-ppb-1.0.0.tar.gz --pre pid=62736
py: commands[0]> make coverage TEST_ARGS=tests
PYTHONPATH="testapp" \
                python -b -W always -m coverage run testapp/manage.py test ${TEST_ARGS:-tests}
/Users/silvag3/github.com/surface-security/django-environ-ppb/.tox/py/lib/python3.11/site-packages/certifi/core.py:36: DeprecationWarning: path is deprecated. Use files() instead. Refer to https://importlib-resources.readthedocs.io/en/latest/using.html#migrating-from-legacy for migration advice.
  _CACERT_CTX = get_path("certifi", "cacert.pem")
/Users/silvag3/github.com/surface-security/django-environ-ppb/.tox/py/lib/python3.11/site-packages/environ/environ.py:754: UserWarning: Error reading /Users/silvag3/github.com/surface-security/django-environ-ppb/testapp/local.env - if you're not configuring your environment separately, check this.
  warnings.warn(
/Users/silvag3/github.com/surface-security/django-environ-ppb/.tox/py/lib/python3.11/site-packages/django/http/request.py:1: DeprecationWarning: 'cgi' is deprecated and slated for removal in Python 3.13
  import cgi
/Users/silvag3/github.com/surface-security/django-environ-ppb/.tox/py/lib/python3.11/site-packages/django/utils/encoding.py:266: DeprecationWarning: Use setlocale(), getencoding() and getlocale() instead
  encoding = locale.getdefaultlocale()[1] or 'ascii'
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
....
----------------------------------------------------------------------
Ran 4 tests in 0.004s

OK
Destroying test database for alias 'default'...
/Users/silvag3/github.com/surface-security/django-environ-ppb/.tox/py/lib/python3.11/site-packages/coverage/control.py:836: CoverageWarning: No data was collected. (no-data-collected)
  self._warn("No data was collected.", slug="no-data-collected")
coverage report
Name                     Stmts   Miss  Cover   Missing
------------------------------------------------------
ppbenviron/__init__.py      69     69     0%   1-101
------------------------------------------------------
TOTAL                       69     69     0%
```

*tl;dr*: Are we missing Django dependency in this project? If not, what's missing in the steps to have Django available for `testapp`?

**Comment by fopina** [comment]
I think the original app did not have django as a dependency because the app itself had no models or anything that required django.

So django would have to be installed only by tox which should already have been I think...

But either way is fine I suppose!

**Comment by gsilvapt** [comment]
> I think the original app did not have django as a dependency because the app itself had no models or anything that required django.
> 
> So django would have to be installed only by tox which should already have been I think...
> 
> But either way is fine I suppose!

Perhaps `tox` API changed and it's no longer installing with the current instructions. I'll try that approach then. Thanks!

**Comment by gsilvapt** [comment]
I realized I misunderstood how tox handles the `deps` block here: https://github.com/surface-security/django-environ-ppb/blob/main/tox.ini#L7-L9


I assumed this was a matrix of versions to install and run, and therefore it was failing. Closing this as a non-fixable because this app doesn't need to directly depend on Django.
